homeassistant/components/klyqa/datacoordinator.py

```diff
@@ -0,0 +1,224 @@
+"""Klyqa datacoordinator."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+import socket
+from typing import Any
+
+from klyqa_ctl import klyqa_ctl as api
+
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_SCAN_INTERVAL,
+    CONF_USERNAME,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_component import (
+    DEFAULT_SCAN_INTERVAL,
+    EntityComponent,
+)
+
+from .const import (
+    CONF_POLLING,
+    CONF_SYNC_ROOMS,
+    EVENT_KLYQA_NEW_LIGHT,
+    EVENT_KLYQA_NEW_LIGHT_GROUP,
+    LOGGER,
+)
+
+
+class HAKlyqaAccount(api.Klyqa_account):  # type: ignore[misc]
+    """HAKlyqaAccount."""
+
+    hass: HomeAssistant | None
+
+    udp: socket.socket
+    tcp: socket.socket
+    polling: bool
+    sync_rooms: bool
+    scan_interval_conf: float
+
+    def __init__(
+        self,
+        udp: Any,
+        tcp: Any,
+        username: str = "",
+        password: str = "",
+        host: str = "",
+        hass: HomeAssistant | None = None,
+        polling: bool = True,
+        sync_rooms: bool = True,
+        scan_interval: float = -1.0,
+    ) -> None:
+        """HAKlyqaAccount."""
+        super().__init__(username, password, host)
+        self.hass = hass
+        self.udp = udp
+        self.tcp = tcp
+        self.polling = polling
+        self.sync_rooms = sync_rooms
+        self.scan_interval_conf = scan_interval
+
+    async def send_to_bulbs(
```

Comment:
Most of  this looks like something that should be implemented in the client Library, not Home Assistant.